SMLHelper/Patchers/CraftDataPatcher_Subnautica.cs

```diff
@@ -1,97 +1,301 @@
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetCraftTimePrefix))));
+            harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.GetCookedData)),
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetCookedCreaturePrefix))));
+            harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.GetBackgroundType)),
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetBackgroundTypesPrefix))));
+            harmony.Patch(AccessTools.Method(typeof(CraftData), nameof(CraftData.IsBuildableTech)),
+               prefix: new HarmonyMethod(AccessTools.Method(typeof(CraftDataPatcher), nameof(GetBuildablePrefix))));
+            
+            IngameMenuHandler.Main.RegisterOnQuitEvent(() => ResetAltered());
 
-            AddCustomTechDataToOriginalDictionary();
         }
 
-        private static void AddCustomTechDataToOriginalDictionary()
+        private static void ResetAltered()
+        {
+            TechDataAltered = 0;
+            HarvestOutputAltered = 0;
+            HarvestTypeAltered = 0;
+            FinalCutBonusAltered = 0;
+            ItemSizesAltered = 0;
+            EquipmentTypesAltered = 0;
+            SlotTypesAltered = 0;
+            CraftingTimesAltered = 0;
+            CookedCreatureAltered = 0;
+            BackgroundTypesAltered = 0;
+            BuildablesAltered = 0;
+            PrefabAltered = 0;
+        }
+
+        private static void CraftDataGetPrefabPrefix()
+        {
+            if (ModPrefab.PrefabCount > 0)
+            {
+                ModPrefab prefab = ModPrefab.Prefabs.FirstOrFallback(null);
+                if (!CraftData.techMapping.ContainsKey(prefab.TechType) || !CraftData.entClassTechTable.ContainsKey(prefab.ClassID))
+                    CraftDataPrefabIDCachePostfix();
+            }
+        }
+
+        private static void CraftDataPrefabIDCachePostfix()
+        {
+            Dictionary<TechType, string> techMapping = CraftData.techMapping;
+            Dictionary<string, TechType> entClassTechTable = CraftData.entClassTechTable;
+            foreach (ModPrefab prefab in ModPrefab.Prefabs)
+            {
+                techMapping[prefab.TechType] = prefab.ClassID;
+                entClassTechTable[prefab.ClassID] = prefab.TechType;
+            }
+        }
+
+        private static void CraftDataGetPrefix()
+        {
+            if (TechDataAltered < CustomTechData.Keys.Count)
+                AddCustomDataToOriginalDictionaries();
+        }
+
+        private static void GetHarvestOutputPrefix()
+        {
+            if (CustomHarvestOutputList.Count > 0 && HarvestOutputAltered < CustomHarvestOutputList.Count)
+            {
+                PatchUtils.PatchDictionary(CraftData.harvestOutputList, CustomHarvestOutputList);
+                HarvestOutputAltered = CustomHarvestOutputList.Count;
```

Comment:
`HarvestOutputAltered` and the various other `...Altered` fields are being assigned with Count values but are being used more like boolean checks.

Is the numeric value of these new fields actually important?  
Or would be be enough to handle them as `bool`s instead?

Reply:
Bool would work better if we added a bool false in the methods that add to those.  Which would have been a far better way to do it.   Will change.

Reply:
Actually ended up going another way altogether Please RECHECK.

Reply:
The new method looks good, but now you've got all those leftover `int` fields to clean up.

Reply:
1790ea9